test/tests/functional/pbs_que_resc_usage.py

```diff
@@ -0,0 +1,180 @@
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            'resources_available.ncpus': 5},
+                            self.mom.shortname)
+        a = {'queue_type': 'execution', 'enabled': True, 'started': True}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='new_q')
+
+    def test_built_in_resc(self):
+        """
+        Test built-in resource and check the assigned resource value
+        after unset and shouldn't set to any negative value
+        """
+        # Unset the resource after job completion
+        a = {'resources_available.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id='new_q')
+        j_attr = {ATTR_queue: 'new_q', 'Resource_List.select': '1:ncpus=3'}
+        j = Job(TEST_USER, j_attr)
+        j.set_sleep_time(20)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, jid)
+        q_status = self.server.status(
+            QUEUE, 'resources_assigned.ncpus', id='new_q', extend='f')
+        self.assertEqual(
+            int(q_status[0]['resources_assigned.ncpus']), 3, self.err_msg)
+        self.server.expect(JOB, 'queue', op=UNSET, id=jid)
+        q_status = self.server.status(
+            QUEUE, 'resources_assigned.ncpus', id='new_q', extend='f')
+        self.assertEqual(
+            int(q_status[0]['resources_assigned.ncpus']), 0, self.err_msg)
+        self.server.manager(MGR_CMD_UNSET, QUEUE,
+                            'resources_available.ncpus', id='new_q')
+        q_status = self.server.status(QUEUE, id='new_q', extend='f')
+        self.assertNotIn('resources_available.ncpus',
+                         q_status[0].keys(), self.err_msg)
+        self.assertNotIn('resources_assigned.ncpus',
+                         q_status[0].keys(), self.err_msg)
+
+        # Unset the resource before job completion
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id='new_q')
+        j = Job(TEST_USER, j_attr)
+        j.set_sleep_time(50)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, jid)
+        q_status = self.server.status(
+            QUEUE, 'resources_assigned.ncpus', id='new_q', extend='f')
+        self.assertEqual(
+            int(q_status[0]['resources_assigned.ncpus']), 3, self.err_msg)
+        self.server.manager(MGR_CMD_UNSET, QUEUE,
+                            'resources_available.ncpus', id='new_q')
+        q_status = self.server.status(QUEUE, id='new_q', extend='f')
+        self.assertNotIn('resources_available.ncpus',
+                         q_status[0].keys(), self.err_msg)
+        self.assertEqual(
+            int(q_status[0]['resources_assigned.ncpus']), 3, self.err_msg)
+        self.server.delete(jid, wait=True)
+        q_status = self.server.status(
+            QUEUE, 'resources_assigned.ncpus', id='new_q', extend='f')
+        self.assertEqual(
+            int(q_status[0]['resources_assigned.ncpus']), 0, self.err_msg)
```

Comment:
Why are you testing this in a second queue?  There is nothing special to the server about workq.  It's just any other queue.  It is created by habitat as the default queue.  Testing a second queue is not necessary.